src/libraries/System.Private.CoreLib/src/System/Numerics/Vector2_Intrinsics.cs

```diff
@@ -214,7 +214,7 @@ public static Vector2 SquareRoot(Vector2 value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Vector2 operator *(float left, Vector2 right)
         {
-            return new Vector2(left, left) * right;
+            return new Vector2(left * right.X, left * right.Y);
```

Comment:
The old way was allocating a `new Vector2(left, left)`, then the `*` operator on `Vector` and `Vector` will also allocate another `Vector2`.

The new way only create one allocation.

Hope that's correct :)

Reply:
What allocation? Vector2 is a struct.

Reply:
Oh. Yes. My fault. I'll close.